plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -141,7 +156,31 @@ export const resolvers = ({
       args: GeneralArgs,
       _context = {}
     ): Observable<JobLink> {
-      return runJob(args.id).map(({ jobId }) => ({ jobId }));
+      if (args.id) {
```

Comment:
What would you think about using a special interface here and using a user-defined type check?

Reply:
I like that idea. But I think it is adding one more level of complexity to something that should not exist. Graphql should check the existence and type of my args, I should not have to add all those guards to the resolver.

I think here I would also make "the same mistake" everywhere and push forward to have our graphql implementation to care about it.

Reply:
Can you explain what you mean with "our graphql implementation to care about it"

Reply:
For Runtime:
The query this resolver resolves states: `runJob(id: String!)`. Which means that the grapqhl implementation should throw an error if `id` is not present. So at runtime, I expect not to have to write an if.

For type checking:
As for the typescript types, I understand and like the elegance of the user-defined type check like `isJobQueryArg()`, but on this particular case I believe it just hides a cast away from me.  

What I really want to right is:
`args: RunJobArgs`

but then IResolvers do not get happy. That I think is the type who needs a tweek to be able to support me to say `args: RunJobArgs`.

So overall I think the `if` here is also the simple solution to those both problems combined.

Reply:
does that argument, solve this issue?